### PR TITLE
rclone-mover yamls

### PR DESCRIPTION
**Pull request by husky-parul**
**Describe what this PR does**
This PR provides the code for the data mover that uses `rclone` to transfer data between the source and destination sites using `volumesnapshot` feature. Until there's an operator to deploy this container, the following steps can be used for experimentation.

**Is there anything that requires special attention?**
N/A

**Related issues:**
N/A

### CODE REVIEW

File: mover-rclone/deployments/mysql/source/job.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              readOnly: true
+            - name: snapshot-data
+              mountPath: /var/lib/mysql/data
+      volumes:
+        - name: rclone-secret
+          secret:
+            secretName: rclone-secret
+        - name: snapshot-data
+          persistentVolumeClaim:
+          claimName: scribe-snap-pv-claim
```
Comment (cooktheryan) [inline mover-rclone/deployments/mysql/source/job.yaml:30]: I believe the error is in regards to the indentation.
```

  volumes:
  - name: rclone-secret
    secret:
      defaultMode: 420
      secretName: rclone-secret
  - name: snapshot-data
    persistentVolumeClaim:
      claimName: scribe-snap-pv-claim
  - name: default-token-h57k8
    secret:
      defaultMode: 420
      secretName: default-token-h57k8
```

Without the spacing

```
  - emptyDir: {}
    name: snapshot-data
  - name: default-token-h57k8
```

Fix should be
```
        persistentVolumeClaim:
          claimName: snapshot-data
```

Comment (husky-parul) [reply mover-rclone/deployments/mysql/source/job.yaml]: Seems like a kube bug. The yaml does not complain about wrong indentation, instead mounts an EMPTY_DIR on the pod.

File: mover-rclone/deployments/mysql/source/job.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+              value: /var/lib/mysql/data
+            - name: RCLONE_CONFIG_SECTION
+              value: aws-s3-bucket
+          volumeMounts:
+            # name must match the volume name below
+            - name: rclone-secret
+              mountPath: /rclone-config/
+              readOnly: true
+            - name: snapshot-data
+              mountPath: /var/lib/mysql/data
```
Comment (cooktheryan) [inline mover-rclone/deployments/mysql/source/job.yaml:31]: Do we want to make this more generic similar to rsync and use /data

**Comment by cooktheryan** [comment]
I think in general do we want to move it from mysql specific items to generic

File: mover-rclone/deployments/mysql/source/volumesnapshot-dynamic.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: scribe-snap
+spec:
+  volumeSnapshotClassName: gp2-csi
+  source:
+    persistentVolumeClaimName: mysql
```
Comment (husky-parul) [inline mover-rclone/deployments/mysql/source/volumesnapshot-dynamic.yaml:9]: @cooktheryan I need to provide the source pvc name. To generalize it I am naming it to `source-pvc`

**Comment by husky-parul** [comment]
@backube/scribemaintainers - Ready for review.

File: mover-rclone/active.sh [hunk 4]
```diff
@@ -22,11 +19,11 @@ RCLONE_FLAGS=(--checksum --one-file-system --create-empty-src-dirs --progress --
 START_TIME=$SECONDS
 case "${DIRECTION}" in
 source)
-    rclone --config "${RCLONE_CONFIG}" sync "${RCLONE_FLAGS[@]}" /data "${RCLONE_CONFIG_SECTION}:${RCLONE_DEST_PATH}"
+    rclone copy "${RCLONE_FLAGS[@]}" "${MOUNT_PATH}" "${RCLONE_CONFIG_SECTION}:${RCLONE_DEST_PATH}"
```
Comment (JohnStrunk) [inline mover-rclone/active.sh:22]: Are you sure you want `copy` and not `sync`? It won't remove deleted files, so the destination will eventually end up with all files that have ever existed.

**Comment by JohnStrunk** [comment]
lgtm
I don't know if @cooktheryan has any other comments.

**Comment by cooktheryan** [comment]
@husky-parul if you are good with taking it out of Draft. All the changes look great

**Review by JohnStrunk — APPROVED** [review]

File: mover-rclone/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+AWS S3 is used as an intermediate storage for high fan-out data replication.
+
+Until there's an operator to deploy this container, the following steps can be
+used for experimentation.
+
+### Prerequiste
+
+Create secret to access AWS S3 account by replacing `<access-key-id>` and
+`<secret-access-key>` in `rclone.conf`
```
Comment (jeffvance) [inline mover-rclone/README.md:17]: nit: also, `region` and `location-constraint` seem to be required, so should this be mentioned here?
And, is it obvious the path to _rclone.conf_?

**Pull request merged**